gazebo_color_plugin/src/color_plugin.cpp

```diff
@@ -0,0 +1,127 @@
+#include <gazebo/common/Plugin.hh>
+#include <ros/ros.h>
+#include <ros/console.h>
+#include "gazebo_color_sensor_plugin/color_sensor_plugin.h"
+
+#include "gazebo_plugins/gazebo_ros_camera.h"
+
+#include <string>
+#include <stdlib.h>
+
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/CameraSensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+#include <std_msgs/Bool.h>
+
+#include <sensor_msgs/Illuminance.h>
+
+#define COLOR_TOLERANCE 30
+#define PIXEL_THRESHOLD 100 //minimum amount of pixels where the sensor returns true
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_SENSOR_PLUGIN(GazeboRosColor)
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Constructor
+  GazeboRosColor::GazeboRosColor():
+  _nh("color_sensor_plugin"),
```

Comment:
Leave 2 spaces at the beginning.

Reply:
Not sure what you mean with this

Reply:
I think Emi means indentation of the lines

Reply:
Yes, what @ernestmc said.